sceneview/src/main/kotlin/io/github/sceneview/scene/TransformableManager.kt

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018 Google LLC All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.sceneview.scene
+
+import android.util.DisplayMetrics
+import android.view.MotionEvent
+import com.google.ar.sceneform.PickHitResult
+import com.google.ar.sceneform.ux.BaseGestureRecognizer
+import io.github.sceneview.scene.BaseTransformationController
+import com.google.ar.sceneform.ux.DragGesture
+import com.google.ar.sceneform.ux.DragGestureRecognizer
+import com.google.ar.sceneform.ux.GesturePointersUtility
+import com.google.ar.sceneform.ux.PinchGestureRecognizer
+import com.google.ar.sceneform.ux.TwistGestureRecognizer
+import io.github.sceneview.node.Node
+
+/**
+ * Handles selection and gestures on ARNode. Replaces functionality provided by
+ * BaseTransformableNode and TransformationSystem in original Sceneform.
+ */
+class TransformableManager(
```

Comment:
Have a look at Filament introduced Pick API for model node selection

Reply:
I guess I could replace the whole CollisionSystem / Collider system with a Filament pick. The only problem seems to be that I only get an integer describing the renderable that was picked as a result. Is there an easier way to get back the node for this renderable ID than traversing from the root node of scene to find it?